google/cloud/aiplatform/metadata/resource.py

```diff
@@ -0,0 +1,300 @@
+            project (str):
+                Project to get this resource from. Overrides project set in
+                aiplatform.init.
+            location (str):
+                Location to get this resource from. Overrides location set in
+                aiplatform.init.
+            credentials (auth_credentials.Credentials):
+                Custom credentials to use to get this resource. Overrides
+                credentials set in aiplatform.init.
+
+        Returns:
+            resource (_Resource):
+                Instantiated representation of the managed Metadata resource.
+
+        """
+
+        try:
+            return cls(
+                resource_name=resource_name,
+                metadata_store_id=metadata_store_id,
+                project=project,
+                location=location,
+                credentials=credentials,
+            )
+        except exceptions.NotFound:
+            logging.info(f"Resource {resource_name} not found.")
+
+    def update(
+        self,
+        metadata: Optional[Dict] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ):
+        """Updates an existing Metadata resource with new metadata.
+
+        Args:
+            metadata (Dict):
+                Optional. metadata contains the updated metadata information.
+            credentials (auth_credentials.Credentials):
+                Custom credentials to use to update this resource. Overrides
+                credentials set in aiplatform.init.
+
+        """
+
+        if not self._gca_resource:
+            raise ValueError("Invalid resource to update")
+
+        gca_resource = deepcopy(self._gca_resource)
+        gca_resource.metadata.update(metadata)
+        api_client = self._instantiate_client(credentials=credentials)
+
+        try:
+            update_gca_resource = self.__class__._update_resource(
+                client=api_client, resource=gca_resource,
+            )
+            self._gca_resource = update_gca_resource
+        except exceptions.NotFound:
+            logging.info(f"Resource to update '{self.name}' not found.")
+
+    @classmethod
+    @abc.abstractmethod
+    def _create_resource(
+        cls,
+        client: utils.MetadataClientWithOverride,
+        parent: str,
+        resource_id: str,
+        schema_title: str,
+        display_name: Optional[str] = None,
+        schema_version: Optional[str] = None,
+        description: Optional[str] = None,
+        metadata: Optional[Dict] = None,
+    ) -> proto.Message:
+        """Create resource method."""
+        pass
+
+    @classmethod
+    @abc.abstractmethod
+    def _update_resource(
+        cls, client: utils.MetadataClientWithOverride, resource: proto.Message,
+    ) -> proto.Message:
+        """Update resource method."""
+        pass
+
+    @classmethod
+    def _extract_metadata_store_id(cls, resource_name, resource_noun) -> str:
```

Comment:
This seems like it can be a staticmethod.

Reply:
Done! thanks!